analysis/src/SharedTypes.ml

```diff
@@ -425,11 +428,18 @@ module Completable = struct
     | CPObj of contextPath * string
     | CPPipe of contextPath * string
 
-  type typedContext = NamedArg of string
+  type namedArg = {
+    (* What the user has already started writing, if anything *)
+    prefix: string;
+    (* Name of the argument *)
+    name: string;
+  }
+
+  type typedContext = NamedArg of namedArg
```

Comment:
You can use inlined records.
Also `NamedArg` is good but does not covey the fact this is an application context. Perhaps specifying that in the comments on the type definition is the way to go.

Reply:
Perhaps the name suggestion will resolve itself once pattern matching contexts are added.